prompt_testing/README.md

```diff
@@ -0,0 +1,241 @@
+2. Include realistic assembly output (you can use Compiler Explorer to generate examples)
+3. Specify expected topics that a good explanation should cover
+4. Test your new cases to ensure they work as expected
+
+### Human Review Workflow
+
+1. Run tests to generate results:
+   ```bash
+   uv run prompt-test run --prompt current --output my_test_results.json
+   ```
+
+2. Review results interactively:
+   ```bash
+   uv run prompt-test review --results-file prompt_testing/results/my_test_results.json
+   ```
+
+3. Analyze review data:
+   ```bash
+   uv run prompt-test analyze
+   ```
+
+## Best Practices
+
+### Test Case Creation
+
+- **Use real examples**: Generate assembly from actual code using Compiler Explorer
+- **Cover edge cases**: Include empty functions, undefined behavior, truncated assembly
+- **Vary difficulty**: Mix beginner, intermediate, and advanced examples
+- **Quality labels**: Mark cases as good/bad examples to test robustness
+
+### Prompt Development
+
+- **Start small**: Test on a subset before running full suite
+- **Iterate quickly**: Make small changes and measure impact
+- **Use version control**: Keep track of what works and what doesn't
+- **Document changes**: Note the reasoning behind prompt modifications
+
+### Evaluation Strategy
+
+- **Combine metrics**: Use both automatic and human evaluation
+- **Regular testing**: Run tests after any prompt changes
+- **Baseline comparison**: Always compare against previous best version
+- **Long-term tracking**: Monitor performance trends over time
+
+## Integration with Main Service
+
+The testing framework uses the same core logic as the main explain service:
+- Same request/response format (`ExplainRequest`/`ExplainResponse`)
+- Same data preparation (`prepare_structured_data`)
+- Same Claude API calls
+- Same token counting and cost calculation
+
+This ensures that test results accurately reflect production performance.
+
+## Extending the Framework
+
+### Adding New Evaluation Metrics
+
+1. Extend `EvaluationMetrics` class in `evaluation/scorer.py`
+2. Add scoring logic to `AutomaticScorer.evaluate_response()`
+3. Update CLI to display new metrics
+
+### Adding New Test Case Categories
+
+1. Create new JSON file in `test_cases/`
+2. Define category-specific expected topics in scorer
+3. Add any category-specific evaluation logic
+
+### Custom Scoring
+
+You can create custom scoring functions:
+
+```python
+from prompt_testing.evaluation.scorer import AutomaticScorer
+
+class CustomScorer(AutomaticScorer):
+    def score_custom_metric(self, response: str) -> float:
+        # Your custom scoring logic
+        return score
+
+scorer = CustomScorer()
+metrics = scorer.evaluate_response(response, expected_topics, difficulty, token_count)
+```
+
+## Troubleshooting
+
+### Common Issues
+
+1. **API Key Issues**: Ensure `ANTHROPIC_API_KEY` environment variable is set
+2. **Missing Test Cases**: Run `python -m prompt_testing.cli list` to see available cases
```

Comment:
Please update to use the `uv` script not python directly. Look for other missed cases too

Reply:
This is now done